connectors/services/sync.py

```diff
@@ -85,7 +86,7 @@ async def _one_sync(self, connector, es):
 
         try:
             # the heartbeat is always triggered
-            connector.start_heartbeat(self.hb)
+            await connector.heartbeat(self.hb)
```

Comment:
For now if job takes too long to run, connectors will not heartbeat until jobs are complete

Reply:
Should we address it @wangch079?

Reply:
Hi @artem-shelkovnikov , I looked deeper, and I'm inclined not to revert the change, because:
1. the previous `start_heartbeat` methods starts a task to periodically send heartbeat
2. there must be a close method for the connector so that the task can be canceled.

https://github.com/elastic/connectors-python/blob/02597f5d8c449ff02839e37b44de47742379a8dd/connectors/byoc.py#L544-L551

Instead, we can have a workaround here:

https://github.com/elastic/connectors-python/blob/52abdd686a07f41c81b160d2693da2937621f084/connectors/sync_job_runner.py#L93-L94

The changes are:

```python
while not self.elastic_server.done():
    self.connector.heartbeat(interval)
    await asyncio.sleep(JOB_CHECK_INTERVAL) 
```

It's much easier and clean. It can also be removed easily once we have the decoupling ready. WDYT?

Reply:
> The changes are:
> 
> while not self.elastic_server.done():
>     self.connector.heartbeat(interval)
>     await asyncio.sleep(JOB_CHECK_INTERVAL) 
> 
> It's much easier and clean. It can also be removed easily once we have the decoupling ready. WDYT?

As long as it works I'm good with it :) 

I'm not sure about the code that you mentioned - will it not block if syncs take too much to execute?

Reply:
>We need to close our resources when we're finishing the sync.

Good call, this is moved to `SyncJobRunner` class. I will add close in the `finally` block

>will it not block if syncs take too much to execute?

That is inside the job execution itself, so it doesn't matter how long the job takes.